packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/utils.dart

```diff
@@ -112,3 +117,263 @@ class BorderLayout extends StatelessWidget {
+    // RenderProperties so that we can animate between those.
+    result.addAll(endRenderProperties.where((prop) => prop.isFreeSpace));
+    return result;
+  }
+
+  @override
+  double get crossAxisDimension => lerpDouble(
+        begin.crossAxisDimension,
+        end.crossAxisDimension,
+        animation.value,
+      );
+
+  @override
+  Axis get crossAxisDirection => end.crossAxisDirection;
+
+  @override
+  List<RenderProperties> crossAxisSpaces(
+      {List<RenderProperties> childrenRenderProps,
+      double Function(Axis) maxSizeAvailable}) {
+    return end.crossAxisSpaces(
+      childrenRenderProps: childrenRenderProps,
+      maxSizeAvailable: maxSizeAvailable,
+    );
+  }
+
+  @override
+  Axis get direction => end.direction;
+
+  @override
+  String get horizontalDirectionDescription =>
+      end.horizontalDirectionDescription;
+
+  @override
+  bool get isMainAxisHorizontal => end.isMainAxisHorizontal;
+
+  @override
+  bool get isMainAxisVertical => end.isMainAxisVertical;
+
+  @override
+  double get mainAxisDimension => lerpDouble(
+        begin.mainAxisDimension,
+        end.mainAxisDimension,
+        animation.value,
+      );
+
+  @override
+  MainAxisSize get mainAxisSize => end.mainAxisSize;
+
+  @override
+  TextBaseline get textBaseline => end.textBaseline;
+
+  @override
+  TextDirection get textDirection => end.textDirection;
+
+  @override
+  num get totalFlex =>
+      lerpDouble(begin.totalFlex, end.totalFlex, animation.value);
+
+  @override
+  String get type => end.type;
+
+  @override
+  VerticalDirection get verticalDirection => end.verticalDirection;
+
+  @override
+  String get verticalDirectionDescription => end.verticalDirectionDescription;
+
+  @override
+  FlexLayoutProperties copyWith(
+      {Size size,
+      List<LayoutProperties> children,
+      BoxConstraints constraints,
+      bool isFlex,
+      String description,
+      num flexFactor,
+      Axis direction,
+      MainAxisAlignment mainAxisAlignment,
+      MainAxisSize mainAxisSize,
+      CrossAxisAlignment crossAxisAlignment,
+      TextDirection textDirection,
+      VerticalDirection verticalDirection,
+      TextBaseline textBaseline}) {
+    return FlexLayoutProperties(
+      size: size ?? this.size,
+      children: children ?? this.children,
+      node: node,
+      constraints: constraints ?? this.constraints,
+      isFlex: isFlex ?? this.isFlex,
+      description: description ?? this.description,
+      flexFactor: flexFactor ?? this.flexFactor,
+      direction: direction ?? this.direction,
+      mainAxisAlignment: mainAxisAlignment ?? this.mainAxisAlignment,
+      mainAxisSize: mainAxisSize ?? this.mainAxisSize,
+      crossAxisAlignment: crossAxisAlignment ?? this.crossAxisAlignment,
+      textDirection: textDirection ?? this.textDirection,
+      verticalDirection: verticalDirection ?? this.verticalDirection,
+      textBaseline: textBaseline ?? this.textBaseline,
+    );
```

Comment:
any reason why this is not AnimatedFlexLayoutProperties?

Reply:
To create a static snapshot of the animation, so that if you trigger a new animation while one is already in progress, there are no jumps between values.